conans/client/conan_api.py

```diff
@@ -366,19 +367,28 @@ def get_test_conanfile_path(tf):
 
     @api_method
     def export_pkg(self, path, name, channel, source_folder=None, build_folder=None,
-                   profile_name=None, settings=None, options=None, env=None, force=False,
-                   no_export=False, user=None, version=None):
+                   install_folder=None, profile_name=None, settings=None, options=None,
+                   env=None, force=False, no_export=False, user=None, version=None):
 
         settings = settings or []
         options = options or []
         env = env or []
 
         cwd = os.getcwd()
         path = self._abs_relative_to(path, cwd)
-        source_folder = self._abs_relative_to(source_folder, cwd, default=cwd)
         build_folder = self._abs_relative_to(build_folder, cwd, default=cwd)
-        profile = profile_from_args(profile_name, settings, options, env=env, scope=None, cwd=cwd,
-                                    client_cache=self._client_cache)
+
+        source_folder = self._abs_relative_to(source_folder, cwd, default=build_folder)
+
+        if not install_folder:
+            profile = profile_from_args(profile_name, settings, options, env=env, scope=None, cwd=cwd,
+                                        client_cache=self._client_cache)
+        else:
+            if profile_name or settings or options or env:
+                self._user_io.out.warn("Ignoring profile/settings/options,"
```

Comment:
Maybe better error and raise, very weird that you provide a --profile that will be simply ignored